crates/typst-library/src/text/font/mod.rs

```diff
@@ -283,6 +310,34 @@ pub struct LineMetrics {
     pub thickness: Em,
 }
 
+/// Metrics for subscripts or superscripts.
+#[derive(Debug, Copy, Clone)]
+pub struct ScriptMetrics {
+    pub width: Em,
+    pub height: Em,
+    pub horizontal_offset: Em,
+    pub vertical_offset: Em,
+}
+
+impl ScriptMetrics {
+    pub const fn default_with_vertical_offset(offset: Em) -> Self {
+        Self {
+            width: Em::new(0.6),
+            height: Em::new(0.6),
+            horizontal_offset: Em::zero(),
+            vertical_offset: offset,
+        }
+    }
+
+    pub const fn default_subscript() -> Self {
+        Self::default_with_vertical_offset(Em::new(-0.2))
+    }
+
+    pub const fn default_superscript() -> Self {
+        Self::default_with_vertical_offset(Em::new(0.5))
+    }
+}
```

Comment:
this would then all be unified with the methods on `ScriptKind`. I think it's also cleaner to not have the defaults live with the element rather than the font parsing code.